File: C++/10845.cpp
#include<iostream>
#include<queue>
using namespace std;
int N;
void solve()
{
    queue<int> Q;
    string cmd;
    int X;
    for(int i=0; i<N; i++)
    {
        cin>>cmd;
        if(cmd == "push")
        {
            cin>>X;
            Q.push(X);
        }
        else if (cmd == "pop")
        {
            if(Q.empty()) cout<<-1<<"\n";
            else
            {
                cout<<Q.front()<<"\n";
                Q.pop();
            }
        }
        else if(cmd == "size")
        {
            cout<<Q.size()<<"\n";
        }
        else if(cmd == "empty")
        {
            if(Q.empty()) cout<<1<<"\n";
            else cout<<0<<"\n";            
        }
        else if(cmd == "front")
        {
            if(Q.empty()) cout<<-1<<"\n";
            else cout<<Q.front()<<"\n";
        }
        else if(cmd == "back")
        {
            if(Q.empty()) cout<<-1<<"\n";
            else cout<<Q.back()<<"\n";
        }
    }
}
void input()
{
    cin>>N;
}
int main()
{
    input();
    solve();
    return 0;
}

File: Python3/10845.py
import sys, queue
N = int(sys.stdin.readline())
Q = queue.Queue()

for _ in range(N):
    cmd = sys.stdin.readline().split()

    if(cmd[0] == 'push'):
        Q.put(cmd[1])
    
    elif(cmd[0] == 'pop'):
        if(Q.empty()):
            print(-1)
        else:
            print(Q.get())
    
    elif(cmd[0] == 'size'):
        print(Q.qsize())
    
    elif(cmd[0] == 'empty'):
        if(Q.empty()):
            print(1)
        else:
            print(0)
    
    elif(cmd[0] == 'front'):
        if(Q.empty()):
            print(-1)
        else:
            print(Q.queue[0])

    elif(cmd[0] == 'back'):
        if(Q.empty()):
            print(-1)
        else:
            print(Q.queue[-1])
